low-level/sensors/impl/generic/node_data.py

```diff
@@ -250,27 +252,14 @@ def _fetch_nw_status(self):
     def fetch_nw_cable_conn_status(self, interface):
         carrier_status = None
         try:
-            carrier_status = self._utility_instance.fetch_nw_cable_status(self.nw_interface_path, interface)
+            carrier_status = Network().get_link_state(interface)
+        except NetworkError as err:
+            logger.debug(err)
+            carrier_status = "UNKNOWN"
         except Exception as e:
-            if e == errno.ENOENT:
-                logger.error(
-                    "Problem occured while reading from nw carrier file:"
-                    f" {self.nw_interface_path}/{interface}/carrier."
-                    "file path doesn't exist")
-            elif e == errno.EACCES:
-                logger.error(
-                    "Problem occured while reading from nw carrier file:"
-                    f" {self.nw_interface_path}/{interface}/carrier."
-                    "Not enough permission to read from the file.")
-            elif e == errno.EPERM:
-                logger.error(
-                    "Problem occured while reading from nw carrier file:"
-                    f" {self.nw_interface_path}/{interface}/carrier."
-                    "Operation is not permitted.")
-            else:
-                logger.error(
-                    "Problem occured while reading from nw carrier file:"
-                    f" {self.nw_interface_path}/{interface}/carrier. Error: {e}")
+            logger.excpetion(
+                "Problem occured while reading from nw carrier file:"
+                f" {self.nw_interface_path}/{interface}/carrier. Error: {e}")

```

Comment:
this exception error string covers the error already ?

Reply:
Yes Chetan, In NetworkError, the logic is for all the OSError Exceptions, so this error log is for any other Exception that is not OSError, for that we are simply printing the Exception error string.

Reply:
> In NetworkError, the logic is for all the OSError Exceptions

I dont see any OSError being handled in NetworkError !?
https://github.com/Seagate/cortx-monitor/blob/main/low-level/framework/platforms/server/error.py#L39

Reply:
while raising the NetworkError.

Reply:
Can you elaborate ?
NetworkError class code reference i gave, there i dont see any OSError being handled, how its happening then

Reply:
```python
## /platforms/server/network.py:get_link_state
except OSError as err:
        raise NetworkError(err.errno, (
            "Failed to read link state for interface '%s' "
            "due to an Error: '%s, %s'"),
            interface, err.strerror, err.filename)
````
We are handling all the OSErrors in the same way. These are not errors for us but a known state of UNKNOWN
Separate error handling of each type of OSError is not required,
OS Errors Handled in previous code, 
1. File Does not exist =>  We don't want to raise error, but status is set to UNKNOWN
2. Not enough permission => will never occur /carrier is read-only file no one can change the file permission
3. Operation not permitted => reading the file is the only operation and it is always supported for /carrier file.

Reply:
Ok i can relate now